tests/views/test_sign_in.py

```diff
@@ -38,9 +47,19 @@ def test_sign_in(self, mock_request):
         self.assertIn("View list of business surveys".encode(), response.data)
         self.assertIn("Sign out".encode(), response.data)
 
+    @requests_mock.mock()
+    def test_sign_in_unable_to_decode_token(self, mock_request):
+        mock_request.post(url_sign_in_data, json={"access_token": 'invalid'}, status_code=201)
+
+        response = self.app.post("/sign-in", follow_redirects=True,
+                                 data={"username": "user", "password": "pass"})
+
+        self.assertEqual(response.status_code, 500)
+        self.assertIn(b'Error 500 - Server error', response.data)
+
     @requests_mock.mock()
     def test_fail_authentication(self, mock_request):
-        mock_request.post(url_sign_in_data, json={"token": "1234abc"}, status_code=401)
+        mock_request.post(url_sign_in_data, json={}, status_code=200)  # No token in response
```

Comment:
This seems to be a test for a failure case, I don't think mocking UAA returning a 200 is appropriate here.

Reply:
UAA could give back a successful response without a token (a test case being tested in backstage). So the success status code makes sense in this case.

Reply:
Sorry misunderstood, I have added the missing 401 UAA response test case and renamed the test for the missing token 👍